modules/core/04-channel/keeper/upgrade.go

```diff
@@ -104,3 +106,107 @@ func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID stri
 		LatestSequenceSend: seq - 1,
 	}, nil
 }
+
+// startFlushUpgradeHandshake will verify the counterparty proposed upgrade and the current channel state.
+// Once the counterparty information has been verified, it will be validated against the self proposed upgrade.
+// If any of the proposed upgrade fields are incompatible, an upgrade error will be returned resulting in an
+// aborted upgrade.
+//
+//lint:ignore U1000 Ignore unused function temporarily for debugging
+func (k Keeper) startFlushUpgradeHandshake(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	proposedUpgradeFields types.UpgradeFields,
+	counterpartyChannel types.Channel,
+	counterpartyUpgrade types.Upgrade,
+	proofCounterpartyChannel,
+	proofUpgrade []byte,
```

Comment:
we should align naming here. Do we want:
```go
proofChannel, 
proofUpgrade, 
```

or 
```
proofCounterpartyChannel, 
proofCounterpartyUpgrade, 
```
?

Reply:
I prefer `proofCounterpartyChannel, proofCounterpartyUpgrade` as it is more explicit and because we do also have our own `proposedUpgrade` in the mix

Reply:
sounds good to me. Will update